src/transactions/transaction.js

```diff
@@ -43,9 +44,12 @@ function createTransaction(recipientId, amount, secret, secondSecret, data, time
 		senderPublicKey: keys.publicKey,
 		timestamp: slots.getTimeWithOffset(timeOffset),
 		asset: {},
-		data,
 	};
 
+	if (data && data.length > 0) {
```

Comment:
As discussed in private, we will now add a check for utf8 string (else throw an error), what gets expected from core. We might think about more universal validation schemes in the future.